### PR TITLE
[purs ide] return documentation comments

### PR DESCRIPTION
**Pull request by nwolverson**
For #2349.

My first iteration used more docs code but I think this is simpler and works out better.

I'm thinking that markdown, plaintext and html options could be provided since there's already a markdown renderer there.

@kRITZCREEK

### CODE REVIEW

File: src/Language/PureScript/Ide/State.hs [hunk 1]
```diff
@@ -246,6 +232,71 @@ resolveLocationsForModule (defs, types) decls =
...
+  convertDecl (IdeDeclarationAnn ann d) = 
+    convertDeclaration'
+      (annotateValue . P.IdentName)
+      (annotateValue . P.IdentName . P.Ident) 
+      (annotateValue . P.TyName . P.ProperName)
+      (annotateValue . P.KiName . P.ProperName)
+      d
+    where
+      docs :: P.Name -> Maybe Text
+      docs ident = convertComments =<< Map.lookup (ident) comments
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Ide/State.hs]: I'd prefer a `fromMaybe ""` at the end of this, so it's easier to tell whether or not we've already attempted to resolve documentation.

File: src/Language/PureScript/Ide/State.hs [hunk 2]
```diff
@@ -246,6 +232,71 @@ resolveLocationsForModule (defs, types) decls =
...
+  Map.mapWithKey (\mn decls ->
+    maybe decls (flip resolveDocumentationForModule decls) (Map.lookup mn modules))
+
+resolveDocumentationForModule
+  :: P.Module
+    -> [IdeDeclarationAnn]
+    -> [IdeDeclarationAnn]
+resolveDocumentationForModule (P.Module _ _ _ sdecls _) decls = map convertDecl decls
+  where 
+  comments :: Map P.Name [P.Comment]
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Ide/State.hs]: I think we need the namespace here, to tell Kinds from Types from data constructors?

File: src/Language/PureScript/Ide/State.hs (hunk 1 above)
Comment (nwolverson) [reply src/Language/PureScript/Ide/State.hs]: So returning `""` instead of `null` in the json output if there is no docs? Maybe `Maybe Maybe _`?

Comment (kRITZCREEK) [reply src/Language/PureScript/Ide/State.hs]: I thought about `Maybe Maybe` but at that point I'd prefer a custom three valued data type. Also note that `convertComments` returns `Nothing` if it sees an empty string.

File: src/Language/PureScript/Ide/State.hs (hunk 2 above)
Comment (nwolverson) [reply src/Language/PureScript/Ide/State.hs]: Is that not inherent in Name? `KiName | TyName | DctorName` etc?

File: src/Language/PureScript/Ide/State.hs (hunk 1 above)
Comment (kRITZCREEK) [reply src/Language/PureScript/Ide/State.hs]: Ahh, I see it now the `KiName` `TyName` ... stuff is from Docs. Sorry for being a little thick xD

While we're at it, there are redundant parens around `ident` in this line.

File: tests/Language/PureScript/Ide/CompletionSpec.hs [hunk 3]
```diff
@@ -6,9 +6,12 @@ import Protolude
 
 import Language.PureScript as P
 import Language.PureScript.Ide.Completion
-import Language.PureScript.Ide.Test
+import Language.PureScript.Ide.Test as Test
+import           Language.PureScript.Ide.Command as Command
```
Comment (kRITZCREEK) [inline tests/Language/PureScript/Ide/CompletionSpec.hs]: The import formatting is inconsistent here, I don't mind leaving space for `qualified`, but let's apply that formatting on every import in this file then.

File: tests/Language/PureScript/Ide/CompletionSpec.hs [hunk 4]
```diff
@@ -21,6 +24,33 @@ reexportMatches =
...
+
+load :: [Text] -> Command
+load = LoadSync . map Test.mn
+
+rebuildSync :: FilePath -> Command
+rebuildSync fp = RebuildSync ("src" </> fp) Nothing
+
+-- spec :: Spec
+-- spec = describe "Rebuilding single modules" $ do
+--     it "rebuilds a correct module without dependencies successfully" $ do
```
Comment (kRITZCREEK) [inline tests/Language/PureScript/Ide/CompletionSpec.hs]: I don't really mind the Type command comment, but this one should be deleted.

Comment (nwolverson) [reply tests/Language/PureScript/Ide/CompletionSpec.hs]: ugh I guess I was tired, such half-assery. thanks

**Comment by kRITZCREEK** [comment]
Great, Thanks!

**Pull request merged**